test/metabase/driver/common/parameters/values_test.clj

```diff
@@ -288,131 +288,132 @@
-                              :filter   [:< [:field $price nil] 3]})
-              expected-sql (str "SELECT "
-                                "\"PUBLIC\".\"VENUES\".\"ID\" AS \"ID\", "
-                                "\"PUBLIC\".\"VENUES\".\"NAME\" AS \"NAME\", "
-                                "\"PUBLIC\".\"VENUES\".\"CATEGORY_ID\" AS \"CATEGORY_ID\", "
-                                "\"PUBLIC\".\"VENUES\".\"LATITUDE\" AS \"LATITUDE\", "
-                                "\"PUBLIC\".\"VENUES\".\"LONGITUDE\" AS \"LONGITUDE\", "
-                                "\"PUBLIC\".\"VENUES\".\"PRICE\" AS \"PRICE\" "
-                                "FROM \"PUBLIC\".\"VENUES\" "
-                                "WHERE \"PUBLIC\".\"VENUES\".\"PRICE\" < 3 "
-                                "LIMIT 1048575")]
-          (mt/with-temp Card [card {:dataset_query mbql-query}]
-            (is (= {:card-id (u/the-id card), :query expected-sql, :params nil}
-                   (value-for-tag
-                    {:name         "card-template-tag-test"
-                     :display-name "Card template tag test"
-                     :type         :card
-                     :card-id      (:id card)}
-                    []))))))))
-
-  (testing "Persisted Models are substituted"
-    (mt/test-driver :postgres
-      (mt/dataset test-data
-        (mt/with-persistence-enabled [persist-models!]
-          (let [mbql-query (mt/mbql-query categories)]
-            (mt/with-temp* [Card [model {:name "model"
-                                         :dataset true
-                                         :dataset_query mbql-query
-                                         :database_id (mt/id)}]]
-              (persist-models!)
-              (testing "tag uses persisted table"
-                (let [pi (db/select-one 'PersistedInfo :card_id (u/the-id model))]
-                  (is (= "persisted" (:state pi)))
-                  (is (re-matches #"select \"id\", \"name\" from \"metabase_cache_[a-z0-9]+_[0-9]+\".\"model_[0-9]+_model\""
-                                  (:query
-                                   (value-for-tag
-                                    {:name         "card-template-tag-test"
-                                     :display-name "Card template tag test"
-                                     :type         :card
-                                     :card-id      (:id model)}
-                                    []))))
-                  (testing "query hits persisted table"
-                    (let [persisted-schema (ddl.i/schema-name {:id (mt/id)}
-                                                              (public-settings/site-uuid))
-                          update-query     (format "update %s.%s set name = name || ' from cached table'"
-                                                   persisted-schema (:table_name pi))
-                          model-query (format "select c_orig.name, c_cached.name
+  (mt/with-test-user :rasta
```

Comment:
This is the only changed line here. It was failing because I changed `segmented-user?` above to default to `true` if no user is bound, along the general principle of failing closed. But that means that we need to run this test in the context of a non-sandboxed user to get it to pass.